Sources/SKTestSupport/Utils.swift

```diff
@@ -51,8 +51,8 @@ public func testScratchDir(testName: String = #function) throws -> URL {
   let url = FileManager.default.temporaryDirectory
     .realpath
     .appendingPathComponent("sourcekit-lsp-test-scratch")
-    .appendingPathComponent(UUID().uuidString)
     .appendingPathComponent(String(testBaseName))
```

Comment:
Are there any concerns about name clashes when running tests in parallel here?

Reply:
You have a point. I was hoping that the test method should be unique but it turns out that we do have two `testMainFilesChanged` methods that will probably both create a scratch directory.

Reply:
Could just add the class name to the name here? Though I suppose even then *technically* we could still clash 😅

Reply:
And how would you do that?

Reply:
Would have to pass it in, ie. `String(describing: Self.self)` or similar.

Reply:
Yes but we don’t have `self` here because this is a top-level function. And I don’t want to add another `self` parameter to `IndexedSingleFileWorkspace` etc. just to get the test class name.

Reply:
Sure, like I said, would have to pass it in. I'm fine if we just want to do UUID instead, though in that case I'd make the test name part of the name with the UUID, rather than a subfolder.

Reply:
I filed https://github.com/apple/sourcekit-lsp/issues/929 to improve test scratch directories in general.